layout/layout.h

```diff
@@ -741,11 +741,11 @@ M0_INTERNAL struct m0_layout *m0_layout_find(struct m0_layout_domain *dom,
  * If buffer is too small then returns the first very lid. If too large then
  * returns the layout id describing biggest possible unit_size.
  *
- * Always returns valid layout id.
+ * Returns -EINVAL if cannot find the right layout.
  */
-M0_INTERNAL uint64_t m0_layout_find_by_buffsize(struct m0_layout_domain *dom,
-						struct m0_fid *pver,
-						size_t buffsize);
+M0_INTERNAL int m0_layout_find_by_buffsize(struct m0_layout_domain *dom,
```

Comment:
should it be something like:
m0_layout_by_objsize_pick()?

Reply:
To be exact, we give the buffer size here, not the object size. The object size may be huge, but the user can select a small buffer size for m0_io_op. (Which is probably wrong, but we can do nothing about it here.)